bluemira/codes/process/_solver.py

```diff
@@ -199,8 +204,139 @@ def execute(self, run_mode: str | RunMode) -> ParameterFrame:
         if teardown := self._get_execution_method(self._teardown, run_mode):
             teardown()
 
+        if self.plot:
+            self.plot_radial_build()
+
         return self.params
 
+    def plot_radial_build(
+        self,
+        width: float = 1.0,
+        *,
+        show: bool = True,
+    ) -> plt.Axes | None:
+        """
+        Plot PROCESS radial build.
+
+        Parameters
+        ----------
+        width:
+            The relative width of the plot.
+        show:
+            If True then immediately display the plot,
+            else delay displaying the plot until
+            the user shows it, by default True.
+
+        Returns
+        -------
+        The plot Axes object.
+        """
+        if not self._teardown:
+            return None
+
+        radial_build = self._teardown._mfile_wrapper.ordered_radial_build
+
+        if not radial_build:
+            bluemira_warn("MFILE.DAT file in old format. Cannot plot radial build.")
+            return None
+
+        R_0 = radial_build["R_0"]
+
+        col = {
+            "Gap": "w",
+            "blanket": "#edb120",
+            "TF coil": "#7e2f8e",
+            "Vacuum vessel": "k",
+            "Radiation shield": "#5dbb63",
+            "Plasma": "#f77ec7",
+            "first wall": "#edb120",
+            "Machine bore": "w",
+            "dr_cs_precomp": "#0072bd",
+            "scrape-off": "#a2142f",
+            "solenoid": "#0072bd",
+            "Thermal shield": "#77ac30",
+        }
+
+        _, ax = plt.subplots(figsize=[14, 10])
+
+        lpatches = []
+        gkeys = [
+            "blanket",
+            "TF coil",
+            "Vacuum vessel",
+            "Radiation shield",
+            "Plasma",
+            "scrape-off",
+            "solenoid",
+            "Thermal shield",
+        ]
+        glabels = {
+            "blanket": "Breeding blanket",
+            "TF coil": "TF coil",
+            "Plasma": "Plasma",
+            "Vacuum vessel": "Vacuum vessel",
+            "Radiation shield": "Radiation shield",
+            "scrape-off": "Scrape-off layer",
+            "solenoid": "Central solenoid",
+            "Thermal shield": "Thermal shield",
+        }
+
+        for comp in radial_build["Radial Build"]:
+            # Generate coordinates for an arbitrary
+            # height radial width.
+            xc = [
+                comp[2] - comp[1],
+                comp[2] - comp[1],
+                comp[2],
+                comp[2],
+                comp[2] - comp[1],
+            ]
+            yc = [-width, width, width, -width, -width]
+            yc = np.array(yc)
+            coords = Coordinates({"x": xc, "y": yc})
+
+            for key, c in col.items():
+                if key.upper() in comp[0].upper():
```

Comment:
Pretty sure this should be changed as below. Also I dont think any of the keys will match atm. 
```suggestion
                if key.upper() in comp[1].upper():
```

Reply:
fixed the issues, so should be reductant